options.go

```diff
@@ -77,6 +84,19 @@ func checkOptions(rp *Ringpop) []error {
 
 // Runtime options
 
+// Clock is used to set the Clock mechanism. Passing nil will use the system
+// clock. Testing harnesses will typically replace this with a mocked clock.
+func Clock(c clock.Clock) Option {
+	return func(r *Ringpop) error {
+		if c == nil {
```

Comment:
The problem is slightly tricky -- both a swim.Node, and ringpop.Ringpop want to be able to have sensible defaults. Options is defined at a level outside swim. Unless we want swim to refer to ringpop.options for its own defaults, using nil allows each of them to do the sane thing here. Perhaps there's a more elegant way of handling this in go, but modularity makes this slightly trickier, afaict.

Modularity aside, I think nil is a valid default for "use the system clock", such that a user then has no need to know how to instantiate one of them, similar to a sane global default.

@CorgiMan @dansimau